osquery/tables/system/posix/sudoers.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <locale>
+#include <vector>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kSudoFile = "/etc/sudoers";
+
+std::vector<std::string> readFromFile(const std::string& path) {
```

Comment:
Is there any particular reason this needs to be in a function? Seems as though all of this work could live in `genSudoers`, and thus you'd avoid having to return a `std::vector<std::string>` which could potentially be quite large.